### PR TITLE
Updating Dashboard Wallet Factory to Allow HD Path Input

### PR DESCRIPTION
**Pull request by ipbyrne**
This PR updates the `dashboardWalletFactory.toKeyPair()` method to allow the HD path parameter. This PR also adds a sanity check test to make sure that the `dashboardWalletFactory.toKeyPair()` and the `@sidetree.wallet's toKeyPair()` operations generate the exact same keys as https://api.did.actor/ when using the same input.

### CODE REVIEW

**Review by OR13 — CHANGES_REQUESTED** [review]
Please fix the arguments and remove the optionality / defaulting.

File: packages/wallet/src/pluginFactory.ts [hunk 1]
```diff
@@ -32,7 +32,8 @@ interface SidetreePlugin {
   toKeyPair: (
     mnemonic: string,
     index: number,
```
[the remark this answers is not in the document]
Comment (ipbyrne) [reply packages/wallet/src/pluginFactory.ts:34]: Copy that!

[the remark this answers is not in the document]
Comment (ipbyrne) [reply packages/wallet/src/pluginFactory.ts]: This has been updated!

File: packages/crypto/src/__tests__/generate-key-fixture.test.ts [hunk 2]
```diff
@@ -26,9 +26,9 @@ it('can generate test fixture', async () => {
 
   for (let i = 0; i < mnemonic.mnemonic.length; i++) {
     const m0 = mnemonic.mnemonic[i];
-    const k0 = await toKeyPair(m0, 0, 'Ed25519');
-    const k1 = await toKeyPair(m0, 0, 'X25519');
-    const k2 = await toKeyPair(m0, 0, 'secp256k1');
+    const k0 = await toKeyPair(m0, 'Ed25519', "m/44'/1'/0'/0/0");
```
Comment (OR13) [inline packages/crypto/src/__tests__/generate-key-fixture.test.ts:29]: much better / safer... just beware of `/1` vs `/0` (BTC) vs `/60` (ETH)

File: packages/crypto/src/mnemonic/toKeyPair.ts [hunk 3]
```diff
@@ -21,14 +21,18 @@ import { SIDETREE_BIP44_COIN_TYPE } from '../constants';
...
 ): Promise<any> => {
   const seed = await bip39.mnemonicToSeed(mnemonic);
   const root = hdkey.fromMasterSeed(seed);
   const addrNode = root.derive(
-    hdPath ? hdPath : `m/44'/${SIDETREE_BIP44_COIN_TYPE}'/0'/0/${index}`
+    hdPath
+      ? index >= 0
+        ? `${hdPath.slice(0, -1)}${index}`
+        : hdPath
+      : `m/44'/${SIDETREE_BIP44_COIN_TYPE}'/0'/0/${index >= 0 ? index : 0}`
```
Comment (OR13) [inline packages/crypto/src/mnemonic/toKeyPair.ts:35]: kill this complexity. get rid of index.

File: packages/wallet/src/pluginFactory.ts [hunk 4]
```diff
@@ -31,8 +31,9 @@ interface SidetreePlugin {
   toMnemonic: (mnemonic?: string) => Promise<Mnemonic>;
   toKeyPair: (
     mnemonic: string,
-    index: number,
-    type?: string
+    type: string,
+    hdPath: string,
+    index?: number
```
Comment (OR13) [inline packages/wallet/src/pluginFactory.ts:36]:
```suggestion
```

**Review by OR13 — CHANGES_REQUESTED** [review]
see comments

File: packages/wallet/src/pluginFactory.ts (hunk 4 above)
Comment (ipbyrne) [reply packages/wallet/src/pluginFactory.ts:36]: Pulling this out now!

File: packages/crypto/src/__tests__/generate-key-fixture.test.ts (hunk 2 above)
Comment (ipbyrne) [reply packages/crypto/src/__tests__/generate-key-fixture.test.ts:29]: Copy that!

File: packages/crypto/src/mnemonic/toKeyPair.ts (hunk 3 above)
Comment (ipbyrne) [reply packages/crypto/src/mnemonic/toKeyPair.ts]: The deed has been done!

**Review by OR13 — APPROVED** [review]

**Pull request merged**